src/js/yeldaWebchatInjector.js

```diff
@@ -278,8 +243,18 @@ class YeldaChat {
     data.shouldBeOpened = data.shouldBeOpened ? true : false
     data.canBeClosed = data.canBeClosed ? true : false
 
-    if (!data.framePosition || validFramePosition.indexOf(data.framePosition) === -1) {
-      data.framePosition = 'outer'
+    // Assign parentContainerId only if the element presents else follow outer behavior
+    if (
+      !data.parentContainerId ||
+      data.parentContainerId === undefined ||
+      data.parentContainerId === null ||
```

Comment:
1) No need of all these checks : if data.something is undefined or null, it will be handled by```!data.parentContainerId```
2) I asked to check for parentContainerId in other function. It should be kept, because createContainer (or other function) should not crashed if called with invalid data. What you could plan is maybe to set this.parentContainer here and reuse it in other functions as ``` this.parentContainer || document.body``` to have a valid parentContainer, and ```this.parentContainer === document.body``` 

``` javascript
// default container
this.parentContainer = document.body 

if(data.parentContainerId && document.getElementById(data.parentContainerId)) {
  this.parentContainer =document.getElementById(data.parentContainerId)
  data.canBeClosed = false
}